src/backend/joanie/payment/backends/base.py

```diff
@@ -59,16 +59,25 @@ def _do_on_payment_success(cls, order, payment):
             reference=payment["id"],
         )
 
+        # triggers flow update
         order.set_installment_paid(payment["installment_id"])
 
-        upcoming_installment = order.state == ORDER_STATE_COMPLETED
+        credit_card_last_numbers = order.credit_card.last_numbers
+        if order_completed := order.state == ORDER_STATE_COMPLETED:
+            credit_card = order.credit_card
+            order.credit_card = None
+            order.save()
+            if not credit_card.orders.exists():
```

Comment:
If there are orders but those one are in state `completed`, `cancel`, `refunding` or `refunded` we could assume to remove this credit_card.